ext/liquid_c/block.c

```diff
@@ -125,6 +125,90 @@ static int is_id(int c)
     return rb_isalnum(c) || c == '_';
 }
 
+typedef struct full_token_possibly_invalid {
+    long body_len;
+    const char *delimiter_start;
+    long delimiter_len;
+} full_token_possibly_invalid_t;
+
+static bool match_full_token_possibly_invalid(token_t *token, full_token_possibly_invalid_t *match)
+{
+    const char *str = token->str_full;
+    long len = token->len_full;
+
+    match->body_len = 0;
+    match->delimiter_start = NULL;
+    match->delimiter_len = 0;
+
+    if (len < 5) return false; // Must be at least 5 characters: \{%\w%\}
```

Comment:
Although it is only called for raw tags at the moment, the name implies it could be used for parsing other tags. So, could it be called within `{% liquid ... %}`? In which case the tags would not be wrapped in `{% ... %}`.